### PR TITLE
Cache requests to Find for 5 minutes

### PR DESCRIPTION
**Pull request by tijmenb**
## Context

We're launching the new cycle tomorrow, and we're expecting higher traffic than normal. One of the first pages that users will hit is the "Apply from Find" page (eg. https://www.apply-for-teacher-training.service.gov.uk/candidate/apply?courseCode=2QW6&providerCode=G48). This page makes a request to the Find API if it can't find the course locally.

## Changes proposed in this pull request

Cache the response for 5 minutes, so we avoid hammering the API if there's a lot of traffic.

Also implements better logging - we'll be able to see outgoing API requests in Logit.

## Guidance to review

Ok?

## Link to Trello card

Not sure, @tvararu ?

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/models/find_api/resource.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module FindAPI
   class Resource < JsonApiClient::Resource
     self.site = ENV.fetch('FIND_BASE_URL')
+    self.connection_options = { headers: { user_agent: 'Apply for teacher training' } }
```
Comment (duncanjbrown) [inline app/models/find_api/resource.rb:4]: 💯

File: app/models/find_api/resource.rb [hunk 2]
```diff
@@ -4,3 +4,24 @@ class Resource < JsonApiClient::Resource
     self.connection_options = { headers: { user_agent: 'Apply for teacher training' } }
   end
 end
+
+require 'faraday/logging/formatter'
+
+class APIRequestLoggingFormatter < Faraday::Logging::Formatter
```
Comment (duncanjbrown) [inline app/models/find_api/resource.rb:10]: Should we namespace this under `FindAPI`?

**Review by duncanjbrown — APPROVED** [review]
LGTM, nit _a volonté_

**Review by tvararu — APPROVED** [review]

**Comment by tijmenb** [comment]
@duncanjbrown je decide de merge la request du pull en ce interést de timing

**Pull request merged**